Stop spamming logs when setting duplicate receiver

cedric-cordenier opened the pull request:
## Ticket
<!---
  Does this work have a corresponding ticket?

  Please link your Jira ticket by including it in one of the following reference:
    - the PR title
    - branch name
    - commit message
  
  By referencing it, it will let the QA team to know what to watch out for when creating a new release.

  Example:

  [LINK-777](https://smartcontract-it.atlassian.net/browse/LINK-777)
--> 

## Requires Dependencies
<!---
  Does this work depend on other open PRs?

  Please list other PRs that are blocking this PR.

  Example:

  - https://github.com/smartcontractkit/chainlink-common/pull/7777777
-->

## Resolves Dependencies
<!---
  Does this work support other open PRs? 

  Please list other PRs that are waiting for this PR to be merged.

  Example:

  - https://github.com/smartcontractkit/ccip/pull/7777777
-->

cedric-cordenier committed:
Stop spamming logs when setting duplicate receiver

cedric-cordenier committed:
Stop spamming logs when setting duplicate receiver

cedric-cordenier committed:
Fix syncer test

bolekk commented on core/capabilities/syncer.go:
```diff
@@ -543,7 +543,12 @@ func (s *registrySyncer) addReceiver(ctx context.Context, capability kcr.Capabil
+	if errors.Is(err, remote.ErrReceiverExists) {
+		// If a receiver already exists, let's log the error for debug purposes, but
+		// otherwise short-circuit here. We've handled this capability in a previous iteration.
+		s.lggr.Debugf("failed to set receiver for cap ID %s and don ID %d: %s", fullCapID, don.Id, err)
```
Maybe "receiver already exists" instead of "failed to set"?

bolekk commented on core/capabilities/syncer.go:
```diff
@@ -543,7 +543,12 @@ func (s *registrySyncer) addReceiver(ctx context.Context, capability kcr.Capabil
 	s.lggr.Debugw("Enabling external access for capability", "id", fullCapID, "donID", don.Id)
 	err = s.dispatcher.SetReceiver(fullCapID, fmt.Sprint(don.Id), receiver)
-	if err != nil {
+	if errors.Is(err, remote.ErrReceiverExists) {
```
Does "Is" work correctly even if it's a longer string with ErrReceiverExists inside it?

cedric-cordenier replied:
It does because we're wrapping the error using the %w directive. The test covers this behaviour

ettec commented on core/capabilities/remote/dispatcher.go:
```diff
@@ -94,7 +99,7 @@ func (d *dispatcher) SetReceiver(capabilityId string, donId string, rec remotety
 	_, ok := d.receivers[k]
 	if ok {
-		return fmt.Errorf("receiver already exists for capability %s and don %s", capabilityId, donId)
+		return fmt.Errorf("%w: capability %s and don %s", ErrReceiverExists, capabilityId, donId)
```
nit: I would expect the caller to wrap/log this error with the capability ID etc (as it is doing), so no need to wrap the error at this point.

cedric-cordenier committed:
Rebase

cedric-cordenier committed:
Address comments

cedric-cordenier committed:
Merge branch 'develop' into logging-follow-up

Tofel committed:
[TT-1128] remove evm client usages (#13568)

* remove EVMClient, ContractDeployer, etc

* remove actions_seth, move them to actions

* fix imports

* decouple docker env builder from evm client creation

* fix vrfv2/plus and lp tests

* fix log poller smoke tests

* remove Seth-related methods moved to CTF

* add fund return to docker tests

* add missing fund returns + cosmetic changes

* update seth readme

EasterTheBunny committed:
verify contract when binding (#13613)

* verify contract when binding

* fix tests that rely on Bind

* fix len check

* change error to custom error

* add error test

amit-momin committed:
Add transaction status function for external components to use (#13040)

* Added transaction status function for external components to use

* Added changeset and fixed linting

* Updated changeset

* Made internal error regex matching explicit

* Updated common client error list

* Defined a generalized tx error interface for chain agnostic compatibility

* Adjusted to align as close as possible with the ChainWriter interface

* Updated to align transaction status method to ChainWriter expectations

* Updated to use the TransactionStatus type from chainlink-common

* Simplified the TxError interface

* Updated FindTxWithIdempotencyKey to return error if no rows found

* Removed sql error checks outside of txstore

* Updated method signature to accept string instead of uuid

* Updated mocks

* Moved tx finality check to Confirmer and renamed fields

* Updated tx finalized check to consider re-orgs

* Fixed linting

* Removed finality from GetTransactionStatus method

* Cleaned out unused client method

* Rephrased comment to match others

jmank88 committed:
.github: reduce headings one level (#13572)

* .github: reduce headings one level

* rm Ticket heading

lukaszcl committed:
Add new OCR tests with RPC node failures (#13624)

* Add ocr soak test with rpc node down for some period

* Remove local replace from go mod tidy

* Cleanup

* Add Grafana annotation if configured

* Refactor

* Do not show error when chaos already deleted

* Bump ctf

* Update chainlink_node_funding

* Add tests to makefile

* Fix lint

* Fail test immediately if chaos cannot be created

* Fix

* Make forwarderFlow an opt

* Fix lint

lukaszcl committed:
Add more tests to ocr soak test workflow (#13629)

DeividasK committed:
SkipFlakey: TestIntegration_MercuryV1 + TestIntegration_MercuryV2_Plugin (#13582)

* SkipFlakey: TestIntegration_MercuryV2_Plugin

* SkipFlakey: TestIntegration_MercuryV1_Plugin

* Mark the right test 🙈

jinhoonbang committed:
relax subscriptionAPI pragma version (#13600)

* relax subscriptionAPI pragma version

* add changeset

Tofel committed:
[TT-1218] try modified compatibility pipeline (#13443)

* try modified compatibility pipeline

* don't wait with matrix generation for image to be built

* fix outputs names

* use images instead of tags

* fix array population

* construct array differently

* fix & debug

* fix echos

* try running tests

* fix path

* fix paths and run command

* more tests + change names + slack notification fix

* test ignored tags, add more tests

* fix slack + more tests

* fix script

* fix ignored tags array

* add vrfv2plus tests, commenout out vrfv2 tests that always fail

* disable vrfv2plus tests for besu and nethermind, modify pipeline triggers

* adjust triggers

* another trigger update

* not trigger should work as expected

* fix again

* yet another fix + add keeper test

* debug 1

* debug 2

* fix output target

* try different condition

* here comes another one...

* fix da condition again, maybe forevah?

* another attempt

* fix yaml

* add go script to check whether new releases created N days ago, update pipeline

* a bit of debug

* fix bash expansion

* return latest tag from go script + get correct ref when building chainlink image depending on the trigger

* fix workflow add depedency

* fix chainlink version that's passed to build

* fix ... mistake

* use correct chainlink image tag in tests

* allow to pass evm implementations and image count in workflow dispatch

* support for custom test/evm implementation/versions input

* fix custom input parsing

* use go script to get latest images, another fix of input decoding

* fix workflow conditions

* further fix the workflow

* remove one more invalid comment

* remove profile from latest_images script

* set aws region from secret

* modify latest images script, so that it ignored tags don't count towards latest images number

* add tests to compatibility script

* refactor latest images and add tests

* add tests for new_release script

* run only Link Billing vrfv2plus test

* fix latest images script

* Add one more test

* fix vrfv2plus regex

* use semver matching when fetching latest images

* fix constraints + warning message

* do not fail slack notification if not all products run

* more tests + variable to force check if image exists

* fix workflow lints

* remove scripts, use CLI tools from CTF

* add go/bin to path

* fix path

* bump go-ethereum to test the pipeline

* bump correct go.mod

* restore go-ethereum version, remove bash script, update compatibility pipeline

* fix pipeline

* remove evm compatiblity json

* add changeset

* fix changeset

* try success() instead of always()

* fix commit selection for workflow_dispatch

* run tests only if build-chainlink succeeds

* WIP: print aggregated results

* try result parsing scripts

* fix needs + script cmd

* run small test

* run small test 2

* fix typos

* do not save empty results

* fix file location + early exit

* do not panic if key is not present

* add concurrency guard

* fix output file name

* fix passing of table

* fix thread link

* fix result printing

* exit early if no results are found

* try wrapping summary table in markdown

* try what happens if there are no results

* fix slack notification part

* remove invalid character

* get results of current workflow

* print results of cron, runlog and flux tests as well + test newer tool version

* do not panic if there no results

* fix aggregated results printing condition

* do not output null, if there are no results

* early exit for asciitable

* remove debug from the pipeline

* check if base64 input is valid

* fix the json validity check

* provide valid json example

* try again with valid example

* fix indentation

* remove whitespaces before and after EOF

* use standalone tools in compat pipeline, modify notify action to also use them

* fix tool name + var name

* fix syntax + use newer asctiitable version

* fix tool name

* save results to input.json

* fix regex passed to notify job

* another slack notification fix

* do not post anything for skipped tests

* do not post, if there are no results

* remove incorrect line

* bump asciitable version

* remove comment

* remove unused action

* adjust live testnets workflow's regex to match new format expected by notify slack job

* bump down go-ethereum in tests

* bump down go-ethereum in core/scripts

lukaszcl committed:
Add docs about E2E chaos tests (#13633)

Tofel committed:
fix live testnet smoke test support (#13634)

Co-authored-by: Ilja Pavlovs <ilja.pavlovs@gmail.com>

skudasov committed:
rename integration tests pipeline (#13631)

jmank88 committed:
core/services/relay: move supported list from common (#13627)

pavel-raykov committed:
Add loggercheck linter to verify that \*w logging methods have even number of args. (#13599)

* Add loggercheck linter to verify that \*w logging methods have even number of args.

* Wrong changes.

* Add .changeset file.

* Add chainlink-common logger linting.

* Remove non-existing logging functions from the linter checks.

* Update the list of logging functions to be linted.

* Remove wrong With linter check.

* Set the .changeset tag to internal.

* Use criticalf, instead of criticalw.

aiagentdev-1990 committed:
KS-307:  Upgrade keystone contracts to 0.8.24 (#13546)

* upgrade keystone contracts to 0.8.24

* update wrappers

* remove submodule

* regen wrappers

* undo pnpm changes

* fix compilation

* update wrappers

* pretify file

* update hardhat config

* upgrade contract versions

* update forwarder logic compiler version

* update wrappers

Tofel committed:
add missing $ sign for some variables (#13645)

RayXpub committed:
`AuthorizedCallers` shared contract (#13625)

* feat: AuthorizedCallers shared contract

* chore: update geth wrappers

* chore: update imported oz version

EasterTheBunny committed:
add limiter for no values to upsert (#13650)

lukaszcl committed:
Add Grafana annotations for OCR chaos tests (#13648)

* Add grafana annotations for ocr tests

* Include test namespace in annotation

* fix go mod

krehermann committed:
fix error scope to avoid data race (#13642)

* fix error scope to avoid data race

* fix another instance, and local scope other errors

* MERC-5697: reinstate test in CI

krehermann committed:
Add forceful testdb creation (#13640)

* Add forceful testdb creation

* fix tests

shileiwill committed:
adding test for v23 (#13644)

* try adding test for v23

* rebase and polish

* address comments

krehermann committed:
move evm db types to evm schema (#13659)

EasterTheBunny committed:
connect background process context to stop channel (#13657)

Tofel committed:
bump CTF to 1.31.4 (#13646)

lukaszcl committed:
Do not log node config in E2E tests (#13666)

ibrajer committed:
Removed obsolete Hardhat VRFv2 tests (#13667)

nolag committed:
Move the bind error test to evm testing so actual chains can run it too (#13652)

nickcorin committed:
evm: Pass value param to the txm (#13602)

* evm: Pass value param to the txm

* changeset: Add changeset

poopoothegorilla committed:
remove error check which skips batching retries [BCI-3359](https://smartcontract-it.atlassian.net/browse/BCI-3359) (#13671)

* remove error check which skips batching retries

* fix typo in changeset

nickcorin committed:
go.mod: Bumped chainlink-common to commit 1eff5dedc9857ed8811186dd2996603942dc1107 (#13670)

* go.mod: Bumped chainlink-common to commit 1eff5dedc9857ed8811186dd2996603942dc1107

* changeset: Add changeset

* mocks: Update mocks

RensR committed:
bump pnpm to v9 (#13630)

* bump pnpm to v9

* fix missing v9 pnpms

* set more restrictive node and pnpm in split tests

tateexon committed:
Cleanup setup-gap errors in gh test summaries (#13654)

chris-de-leon-cll committed:
Remove L2EP HH Tests (#13660)

* remove hardhat tests

* revert expected gas test values

bolekk committed:
[KS-354] Load test for Streams Trigger (#13643)

decoding + validation + aggregation

leeyikjiun committed:
vrfv2: Add request id to logs in runPipelines (#13172)

archseer committed:
Pass capability registry into relayer init (#13663)

makramkd committed:
fix go-plugin version (#13681)

match the version in integration-tests/go.mod

Tofel committed:
[TT-1209] use the same CL node config in k8s as in docker (#13411)

* play around

* remove hardcoded configs for k8s, use TOML test config instead

* adjust Makefile

* fix ocr forwarders panic

* add function for getting correct ocr config for soak tests

* lower chainlink node funding to something reasonable

* warn when no events were collected

* disable forwarder ocr soak tests until fixed

---------

Co-authored-by: Anirudh Warrier <12178754+anirudhwarrier@users.noreply.github.com>

cedric-cordenier committed:
Add standalone registrysyncer package (#13668)

* Add standalone registrysyncer pkg

- And workflow handler implementation for keystone-specific logic

* Linting

* Add test for LocalNode

* Add debug logging

* Rename handler -> launcher

* Remove read lock in Start

* More comments

aalu1418 committed:
fix: solana e2e test compilation (#13689)

Tofel committed:
fix namespace creation in ocrv2 soak tests (#13690)

samsondav committed:
Support "dummy" relayer type (#13583)

* Implement new relayer type "dummy"

This allows testing of plugins without needing a real connection to any
chain.

* Add changeset

* Spelling fix

* Remove useless file

* Update core/services/relay/dummy/llo_provider.go

Co-authored-by: Jordan Krage <jmank88@gmail.com>

* Fix typo

* Move onchainconfig into llo package

---------

Co-authored-by: Jordan Krage <jmank88@gmail.com>

krehermann committed:
re-enable v2 test (#13656)

RyanRHall committed:
remove stale automation HH tests (#13676)

momentmaker committed:
add ci-protobuf gha workflow (#13677)

* add ci-protobuf gha workflow

* fetch-depth: 0 checkout

* add env REPO_URL and BASE_BRANCH

* use https

* use buf actions

* add github_token to action and change .proto file for test

* fix codeowner bug

* fix

* add more explicit *.proto files

* move catch-all foundations owner

* another try

* revert back to original

* revert test

nickcorin committed:
[BCI-3381] Implement `GetTransactionStatus` for the EVM `ChainWriter`. (#13691)

* evm: Implement get transaction status

* evm: Implement get transaction status

* evm: Fix linting issues

leeyikjiun committed:
vrfv2plus: Add batch coordinator tests (#13497)

* vrfv2plus: Add batch coordinator tests

* Add function comment for print generate proof v2 plus

HenryNguyen5 committed:
keystone/workflow engine logging (#13563)

* Remove unused logger arg

* Make capability id logging consistent

* Clean up init logs

* Normalize executionID and capabilityID

* Adding store level logging

* Add changeset

* Add custom errors to workflow engine

* Remove wrapped variants of logging

* Simplify error handling for workflow engine errors

* Change indentation

* Run linter

* Remove trailing newline

Tofel committed:
fix double namespace creation in OCR soak tests (#13696)

* remove merge artifact

* embed more configs

* fix embedded file name

jmank88 committed:
bump mockery to v2.43.2 (#13594)

* bump mockery to v2.43.2

* make generate

jmank88 committed:
core/services/cron: add evmChainID field (#13619)

amit-momin committed:
Updated TXM test for better cleanup (#13698)

patrickhuie19 committed:
adding new ccip custom provider methods (#13699)

* adding new ccip custom provider methods

* bumping cosmos

Tofel committed:
[TT-1304] try CTF with newer eth genesis generator and prysm versions (#13669)

* try CTF with newer eth genesis generator and prysm versions

* add setup-gap id

* use tagged CTF

RensR committed:
rm hardhat split tests (#13682)

* rm hardhat split tests

* cleanup

cedric-cordenier committed:
Allow flat interpolation (#13601)

* Allow flat interpolation

* Update common

* Fix tests after rebase

Tofel committed:
update OCR test names in on-demand workflow (#13708)

jmank88 committed:
bump operator-ui for CronSpec.EVMChainID (#13701)

DeividasK committed:
Simplify capability configuration contract interface check (#13707)

cedric-cordenier committed:
[KS-347] Start tracking registerTrigger goroutine (#13688)

* [KS-347] Start tracking registerTrigger goroutine

* Check for channel closed

* Check for channel closed

---------

Co-authored-by: Bolek <1416262+bolekk@users.noreply.github.com>

Tofel committed:
[TT-1190] support multiple configuration names (#13649)

* support multiple configuration names

* TT-1090: adding code to investigate issue with incorrect config being used

* TT-1090: adding code to investigate issue with incorrect config being used

* TT-1090: adding toml test config for remaining chains for VRF v2 Plus

* support unnamed configuration

* use chain-specific config in VRF tests

---------

Co-authored-by: Ilja Pavlovs <ilja.pavlovs@gmail.com>

cedric-cordenier committed:
Bump common; update mapstructure to incorporate perf improvements (#13717)

aalu1418 committed:
bump: solana with logging fix (#13710)

Tofel committed:
Fix log poller test early exit (#13719)

* WIP#1 - doesn't hang on error

* WIP#3 - better early exit

* use concurrent executor in log poller tests

Tofel committed:
[TT-1167] try updated grafana shortener (#13706)

* support multiple configuration names

* TT-1090: adding code to investigate issue with incorrect config being used

* TT-1090: adding code to investigate issue with incorrect config being used

* TT-1090: adding toml test config for remaining chains for VRF v2 Plus

* support unnamed configuration

* try updated grafana shortener

* use new summary printer in correct job

* use fixed version of print summary action

* another test of shortener

* remove debug

---------

Co-authored-by: Ilja Pavlovs <ilja.pavlovs@gmail.com>

bolekk committed:
[KS-354] Load/integration test for streams consensus aggregator (#13705)

nickcorin committed:
[BCI-3438] Add the chainwriter to the EVM relayer impl (#13718)

* evm: Add chain writer constructor onto the evm relayer impl

* changeset: Add changeset

patrickhuie19 committed:
Adding foundations to root (#13559)

* renamed: CODEOWNERS -> .github/CODEOWNERS

* adding foundations to root

* fixing root path

amit-momin committed:
Update L1 gas price calculations for Optimism Ecotone and Fjord upgrades (#13661)

* Updated L1 gas price calculations for Optimism Ecotone and Fjord upgrades

* Added changeset

* Fixed linting

* Removed panics from L1 oracle initialization

* Fixed linting

* Updated comments

* Fixed upgrade check batch result parsing

* Fixed linting errors

dhaidashenko committed:
Feature/bci 2996 local finality violation (#13048)

* implement RPCClient changes

* simplify interception of RPC's view on chain

* Remove redundant changes

* remove redundant changes

* do not pick RPC as active if it's lagging on finalized block

* fix race

* minor refactor

* Added FinalizedBlockOffset to config

* HeadTracker support of FinalizedBlockOffset

* fix lint issues

* switch logpoller to head tracker

* remove redundant changes

* fix build

* Capture latest chain info on RPC level to avoid race on state transition if we are communicating with RPC with highest finalized block

* EnforceRepeatableRead config option

* ensure HeadTracked does not miss block if it's committed  between initialHead processing and subscription

* hot fix flakey test

* fix build & changeset

* remove outdated test

* fix headtracker tests

* use polling in head tracker to get latest finalized block

* fix relay

* regen mocks

* use proper context in the LogPoller's latestBlocks

* fix flaky test

* fixes for changes from develop branch

* replace panic with FailNow

* report dead node only if it's dead for long enough

* fix racy test

* rollback foundry change caused by merge with develop

* use active node as source of HighestChainInfo to reduce number of transitions to nodeStateFinalizedBlockOutOfSync

* optimised loading of latest finalized block

* fix merge conflicts

* make generate

* simplify capturing of app layer observations

* ensure logpoller is not affected by FinalityTagBypass

* fix comment

* reset keystone to develop

* reset keystone

* reset to develop

* ungoimport

* move newChainIDSubForwarder into the test

* fix typo

* Move deathDeclarationDelay to config

* Comments adjustments

* fix merge issues

* regen mocks

* fix multinode flaky test

* refactor SetTotalDifficultyIfGt to MaxTotalDifficulty

* Use softer language in LatestFinalizedBlock comments

Co-authored-by: Dimitris Grigoriou <dimitris.grigoriou@smartcontract.com>

* nits

* Ensure HistoryDepth is >= FinalizedBlockOffset

* regen mocks with newer version

---------

Co-authored-by: Dimitris Grigoriou <dimitris.grigoriou@smartcontract.com>
Co-authored-by: Domino Valdano <domino.valdano@smartcontract.com>

jmank88 committed:
add make test-short for running short tests with quiet logs (#13628)

aalu1418 committed:
fix: solana e2e test ws error (#13728)

* fix: solana e2e test ws error

* bump to merged commit

bolekk committed:
[KS-363] Update Streams codec to new interface (#13725)

Separated Unwrap() and Validate().

yongkangchia committed:
[BCI-2953] Initial Aptos keystore (#13564)

* chain type to include aptos

* aptos key ed25519 implementation

* add aptos to keystore master

* added exports for keys

* removed sdk dependency

* updated go mod

* Added tests for account sdk deps

* added aptos keystore

* added interface assertion to solana

* added aptos keyring

* added aptos keystore mocks

* added aptos key controller & router

* Added aptos keys command

* added aptos keystore e2e tests

* Added Aptos network ORM test for key

* Added aptos ocr2 keys support

* Revert "Added Aptos network ORM test for key"

This reverts commit 444fe9509647db003b501abd2d8cb72941eea141.

* Added Aptos export functionality

* Modified Aptos Accounts for key rotation

* added aptos keystore to orm validation

* Addded Change logs

* Updated keys implementation to remove address

* Revert "updated go mod"

This reverts commit a8f7e84c8b868eda256f71b576251ce1e76247b6.

* Update changeset

* Updated GraphQL to include Aptos

* Linting fixes

* added aptos configs for keystore

* Changed Aptos config to RawConfig

* Added Mock file for general config

* using *bool type for aptos enabled

* added debug for error

* Fixed tests + added ORM

* Added Test Scripts

* Updated type of raw config

* update all packages go mod tidy

* update all packages go mod

* Update core/services/chainlink/config.go

Co-authored-by: Jordan Krage <jmank88@gmail.com>

* Removed Address in key

* Fixed merge conflicts

* nit changes based on reviews

* Added typecast check for keyring

* Added config type cast check

* make generate update version

* update validation method

* update config validation

* Removed Pointer type for RawConfig

* Updated Validate with better error message

* Added Error config

* Update struct of RawConfig

* removed uncessary config validation

* Added missing starknet keystore tests

* added styling feedback

* Update core/services/keystore/keys/aptoskey/account.go

Co-authored-by: Jordan Krage <jmank88@gmail.com>

* Removed empty array table for config

* fixed lint

---------

Co-authored-by: Jordan Krage <jmank88@gmail.com>

anirudhwarrier committed:
fix benchmark test namespace name (#13729)

Tofel committed:
run tests only if there's > 1 eth implementations or base64 input is not empty (#13738)

snehaagni committed:
[RE-2665] chore/release 2.13.0 to develop (#13737)

* Bump version and update CHANGELOG for core v2.13.0

Signed-off-by: Sneha Agnihotri <sneha.agnihotri@smartcontract.com>

* Curate changelog, remove #internal changesets

Signed-off-by: Sneha Agnihotri <sneha.agnihotri@smartcontract.com>

* Finalize date on changelog for 2.13.0

Signed-off-by: Sneha Agnihotri <sneha.agnihotri@smartcontract.com>

---------

Signed-off-by: Sneha Agnihotri <sneha.agnihotri@smartcontract.com>

ferglor committed:
Only encode non nil block numbers (#13726)

* WIP specify block 0

* Remove 0 block check

* Add changeset

* Tweak changeset

* Only set non nil block numbers

* Lint

cedric-cordenier committed:
[KS-285] Update DON struct; pass through DON Config Version (#13739)

* Convert donID string -> uint32 to match capabilities registry

* Fix launcher tests

* Pass through ConfigVersion

* Update tests

* bump common

jmank88 committed:
.github/CODEOWNERS: add releng and foundations to go.mod (#13747)

shileiwill committed:
support native token (#13714)

* support native token

* split the tests and add mercury related tests

samsondav committed:
Bump chainlink-data-streams to 761f63e7b52700827196b6011401950035317e70 (bid/ask invariant logic on report (#13741)

jmank88 committed:
bump libocr to 063ceef8c42eeadbe94221e55b8892690d36099a (#13746)

patrickhuie19 committed:
downgrading wsrpc to v0.7.3 (#13745)

* downgrading wsrpc to v0.7.2

* bumping to v0.7.3

KodeyThomas committed:
[SHIP-2208] Enable EIP-1559 Transactions on Scroll (#13687)

* Enable EIP-1559 Transactions on Scroll

* docs

* Update to 1 gwei bump min

* Update BlockHistorySize

* docs

* Reduce BlockHistorySize

aalu1418 committed:
disable evm on solana e2e tests to reduce flakinesss (#13744)

* disable evm on solana e2e tests

* bump to merged commit

jmank88 committed:
use services.Ticker (#12668)

* use services.NewTicker

* adjust timing to fix flake

* revert MaxWaitTimeForEvents

vreff committed:
Remove ocr2vrf (#13472)

* Remove ocr2vrf

* Fix merge conflicts

* Update core/services/ocr2/delegate.go

* Remove dkg-related items

* Fix tests

* Fix wrappers gen

* fix help file

* Add changeset

* fix merge conflicts

* fix merge conflicts 2

snehaagni committed:
Bump version and update CHANGELOG fore core v2.14.0 (#13753)

Signed-off-by: Sneha Agnihotri <sneha.agnihotri@smartcontract.com>

davidcauchi committed:
[QA-1378] Remove test config node config override (#13759)

* Remove test config node config override

* Add default toml for 1337

skudasov committed:
fix load test for CRIB, update docs (#13764)

* fix load test for CRIB, update docs

* update docs

* review fixes

ferglor committed:
Dequeue minimum commitment logs for all upkeeps as a priority (#13653)

* Track how many logs we dequeue for each block window

* When the minimum dequeue commitment has been met for all blocks, perform best effort

* WIP tests

* Add iteration step

* Clean up reorg blocks, lock whole enqueue process

* Add tests

* Refactor the dequeue loop into two functions

* Update the log limit of arbitrum to 2, update tests

* Go imports

* Lint

* Add changeset

* Clean up upkeep selector, drop iteration logic

* Clean up unused log limit low

* Update changeset

* Clean up default selector

* Set lookback blocks to 128

* Add comments

* Update comments

* Update tests

* Clean up dequeued counts

* Update logs

* More accurate logs

* Tidy up logging

* - Add log limit to buffer opts
- Use a default log limit of 1
- Simplify dequeue interface

* Clean up based on start window

* Dequeue the smallest of either capacity or remaining log limit

* Evict reorgd blocks outside of blockStatistics

anirudhwarrier committed:
Fix NodeConfig for Automation tests (#13766)

* enable PrometheusPostgresExporter in automation loa test

* remove Database config overrides from automation, keepers BaseConfigTOML

dhaidashenko committed:
Fix TestHeadTracker_CallsHeadTrackableCallbacks flaky test (#13712)

cedric-cordenier committed:
[fix] Make registry satisfy interface (#13773)

* [fix] Make registry satisfy interface

* Update go.mod

Atrax1 committed:
Bcf 3281 ability to deploy node using docker compose (#13721)

* feat(docker): upgrade to new docker compose version

* fix(docker): update name to avoid duplicate with core node

* feat(docker): remove unused env vars

* feat(docker): enable dev mode with rebuild + cleanup

* feat(docker): cleanup and update README

* doc(docker): update README

* doc(docker): update README missing punctuation

aalu1418 committed:
e2e: skip setting EVM networks on nodes in nonEVM e2e tests (#13768)

* skip setting EVM networks in node config for nonEVM e2e tests

* bump solana

* bump solana to merged commit

nolag committed:
Start N workers and have them poll instead of waiting for a worker when the task is ready.  This removes the possibly of an unlucky step being starved if the box is busy (#13767)

samsondav committed:
Add 'stream' to allowed job type list; fix issue adding large numbers of channels (#13692)

* Add 'stream' to allowed job type list

- Improve streams logging; add "Verbose" option
- Bump chainlink-data-streams
- Fix some LLO bugs

* Add changeset

* Add changeset tag

* fix linter

* Latest chainlink-data-streams master

Tofel committed:
[TT-1387] log warning if any custom network config for CL node is present (#13775)

* log warning if any custom network config for CL node is present

* Update integration-tests/testconfig/testconfig.go

Co-authored-by: Adam Hamrick <adam.hamrick@smartcontract.com>

---------

Co-authored-by: Adam Hamrick <adam.hamrick@smartcontract.com>

ibrajer committed:
VRFV2Plus L1 fee calculation updates for Optimism and Base (#13335)

* VRFV2Plus wrapper L1 fee calculation updates for Optimism and Base

* Fixed tx revert in the wrapper deploy universe superscript

* Contract fixes and new superscript command

* Updated pnpm lockfile

* Fixed Optimism L1 transaction padding bytes size

* Added L1 fee calculation options to coordinator and wrapper

* Updated gethwrappers

* Reverted s_coordinatorLinkPremiumPercentage back to private

* Added more comments for different L1 fee calculations

* Added coefficient to L1 gas calculation

* Removed obsolete superscript command

* Updated pnpm lockfile

* Solhint fixes

* Fixed wrapper revert issue

* Fixed superscript linter error

* Another superscript fix

* Optimize remove consumer gas and bytecode. Remove redundant operations

* Revert calldata to memory for several functions in VRFCoordinator

* Fixed all nits and reverted ChainSpecificUtil

* Removed unnecessary deps from package.json and reduced compiler warnings

* Brought back semver package (removed by accident)

* Move L1_UNSIGNED_RLP_ENC_TX_DATA_BYTES_SIZE higher up

* Foundry tests for new L2 chain VRF coordinators

* Diverged wrapper contract and introduce L1GasFee event

* Add tests to check emitting L1GasFee event

* Wrapper tests and fix for L1 gas calculation setter on the wrapper

* Fixed getBlockhash Foundry test for Arbitrum coordinator

* Addressed more comments, minor contract fixes and test updates

* Updated VRF superscripts to support diverged contracts

* Updated Hardhat config and go_generate script with new contracts

* Trimmed down vendor code around Optimism GasPriceOracle contract

* Update proof and rc from memory to calldata

* Update gethwrappers

* Fixed CI issues with formatting and gethwrapper generator

* Updated VRFv2 gethwrappers

* Fixed TestIntegrationVRFV2 test failure

* Fixed failing Foundry tests

* Fixed broken hardhat config

* BatchVRFCoordinatorV2Plus Foundry test fixed and added changesets

---------

Co-authored-by: YikJiun Lee <yikjiun.lee@smartcontract.com>
Co-authored-by: app-token-issuer-infra-releng[bot] <120227048+app-token-issuer-infra-releng[bot]@users.noreply.github.com>

lukaszcl committed:
Load E2E test secrets from env vars  (#13762)

* Load test config secrets from env vars

* Remove secrets from test config in automation-ondemand-tests.yml

* Pass config override

* Update on-demand-vrfv2plus-eth2-clients-test.yml to accept test_secrets_override_key

* Bump gha

* bump ctf

* Use debug in automation workflow test log

* Update input name

* Fix integration-tests.yml

* Override with env vars after overriding with test config

* Clean up setup-create-base64-config action

* bump ctf

* bump ctf and gha

* Update env var names to make it easier to set by QA

* Add .secrets to gitignore

* Load secrets from testconfig/.secrets

* Fix MissingImageInfoAsError

* Load secrets from env file in home dir

* Add checkSecretsInToml

* Update checkSecretsInToml

* Update checkSecretsInToml

* Revert on-demand-vrfv2-eth2-clients-test.yml

* Revert on-demand-vrfv2plus-eth2-clients-test.yml

* Fix go mod

* Fix

* Bump run-tests gha

* bump ctf

* test

* bump gha and ctf

* Fix workflows that use new GHA

* Fix

* Fix lint

* Fix log level

* Use E2E_TEST_ prefix to forward env vars to remote runner

* Add docs

* Fix and bump

* bump ctf and gha

Tofel committed:
use CTF v1.31.10 that doesn't use pod disruption budget in k8s (#13788)

0xsuryansh committed:
feat: Added library for sorted set validations (#13774)

nickcorin committed:
relay: Add chain writer method onto the dummy relayer (#13789)

krehermann committed:
BCF-3276: refactor migrations to use goose provider (#13678)

* BCF-3276: refactor migrations to use goose provider

* linter, cleanup

* changeset

---------

Co-authored-by: Vyzaldy Sanchez <vyzaldysanchez@gmail.com>

cedric-cordenier committed:
[KS-372] Lazy initialize the registrysyncer (#13787)

agusaldasoro committed:
Make send signatures in Transmit configurable (#13761)

* Make send signatures in Transmit configurable

* Add changelog

* fix lint

* Fix tests

* Update .changeset/hungry-pandas-suffer.md

Co-authored-by: Jordan Krage <jmank88@gmail.com>

---------

Co-authored-by: Jordan Krage <jmank88@gmail.com>

Atrax1 committed:
feat(docker): remove specify obsolete version (#13786)

erikburt committed:
ci: downgrade and different runners for ci-core matrix (#13784)

* ci: downgrade and different runners for ci-core matrix

* fix: use 32cores for core_race, 64 for cron runs

* fix: add back short tests to core_tests

* fix: remove short tests step due to no caching

* fix: use ubuntu22.04 instead of ubuntu-latest, formatting

tateexon committed:
[TT-1252] Correct the evm/relayer replace and enable more reuse in the ci-core setup actions (#13592)

* [TT-1252] Evm dispatch

* Bump actions to get evm relayer deps change

* change back to a replace until the migration is complete

* one more replace

* Move the dependency change higher up

* bump chainlink-github-actions versions

* merge conflict fixes

* set specific setup-github-token version

* Add an input base path to the setup actions

* merge conflict fix and bump to newer version

* revert mod version bump since it has a regression

cedric-cordenier committed:
[KS-373] Call bind in Execute() (#13800)

iljapavlovs committed:
VRF-1134: Update super scripts for BHS backward mode to include Nexon updated chain ids (#13792)

nickcorin committed:
go.mod: Bumped versions of other chain repos (#13793)

* go.mod: Bumped versions of other chain repos

* go.mod: Bump starknet version to 03612098f79912876f0f07ea686b279ee5bd865c

* go.mod: Bump starknet relayer version to 03612098f79912876f0f07ea686b279ee5bd865c

* Bump cosmos

* Run make generate

* Add NewChainWriter() to MockRelayer and TestRelayer

---------

Co-authored-by: ilija <pavlovicilija42@gmail.com>
Co-authored-by: ilija42 <57732589+ilija42@users.noreply.github.com>

silaslenihan committed:
Added small check to allow for nil TxMeta in CW SubmitTransaction (#13805)

* Added small check to allow for nil TxMeta in CW SubmitTransaction

* Added changeset

iljapavlovs committed:
VRF-1134: Adjust BHS test to run on chains where txs needed for block… (#13819)

* VRF-1134: Adjust BHS test to run on chains where txs needed for blocks to be generated

* VRF-1134: fixing tests

cedric-cordenier committed:
[chore] Logging improvements and fix to capabilities registry script (#13817)

* Ensure workflow DON is public

* Name loggers

* Log out registry state received by launcher

Tofel committed:
[TT-1376] more stable compatibility pipeline (#13720)

* increase cl node startup timeout

* use newer print summary action

* use /readyz endpoint not /health, when waiting for CL node to be ready

* slightly increase min time between emitting events to avoid 'alreadyKnown' tx error, when processing is slow

* add step that gathers resource metric

* move where telemetry action runs

* use latest CTF

* put collection in correct place, really

* WIP#1 - doesn't hang on error

* WIP#3 - better early exit

* use concurrent executor in log poller tests

* use stronger runners

* use newer CTF

* use free runners, limit parallel to 30

* try with 10x max parallelism

* do not gather workflow metrics

* notify bix-scale-eng about compat failures, use latest tagged ctf

DeividasK committed:
Go: 1.21.7 => 1.22.5 (#13820)

* Go: 1.21.7 => 1.22.5

* Update chainlink dockerfile

* Update dockerfiles

* Add changeset

* Update readme

iljapavlovs committed:
VRF-1137: Include Op stack Coordinator and Wrapper contracts in e2e tests when executing tests in OP stack chains (#13826)

ilija42 committed:
[BCF-3247] - Add confidence levels to CR GetLatestValue (#13514)

* Initial Add confidence levels to CR GetLatestValue

* Implement confidence levels to CR GetLatestValue method contract call

* Add head tracker to CR for head finality info for contract calls

* Setup CR tests for testing GetLatestValue confidence levels

* Resolve methodBinding TODO in blockNumberFromConfidence

* Update common and feeds and resolve minor issues from CI checks

* Implement ClientWithContractHistory to test CR finality handling

* Bump common and feeds

* Bump common and feeds refs

* Bump solana ref

* Skip flaky test

erikburt committed:
fix: sonarscan ci error (#13830)

* fix: bump sonarsource/sonarqube-scan-action to v2.3.0

* fix: increase memory for sonarscan action

* fix: even more memory

* fix: disable coverage redesign

* Revert "fix: disable coverage redesign"

This reverts commit 61557940da5b21c202b5e4aa372fbfe9e9f92c85.

* fix: even more more memory

tateexon committed:
[TT-1252] Make tools/bin/go_core_* tools reusable from other repositories (#13827)

* [TT-1252] Make tools used in ci reusable for chainlink-integrations

* Make fuzz reusable in other reops

* Make the fuzz python arg for the root of the go project more readable as to its meaning

Tofel committed:
use updated CTF (#13828)

* use updated CTF

* use tagged CTF

* verify commit

* fix compile error

---------

Co-authored-by: Anirudh Warrier <12178754+anirudhwarrier@users.noreply.github.com>

nickcorin committed:
[BCI-3813] Use the `transactionID` as the idempotency key in the EVM chain writer (#13829)

* evm: Use the transactionId as the idempotency key in the chainwriter

* changeset: Add changeset tag

---------

Co-authored-by: Silas Lenihan <32529249+silaslenihan@users.noreply.github.com>

ilija42 committed:
[BCF-3269] - Contract Reader Batch Call  (#13635)

* Chain Reader Batch Call inital commit

* Add batchCaller tests

* Minor code improvement for ChainReader addDecoderDef and addEncoderDef

* Use common types for Chain Reader BatchGetLatestValue

* CR BatchGetLatestValue fixes and setup evm chain reader tester

* Run lint, generate and update common ref

* Add changeset

* Update common ref, rename BatchGetLatestValue to BatchGetLatestValues

* Import new common changes

* Return an error if BatchGetLatestValues received an event in the req

* Bump common

* fix minor merge issues

* Add common codec mock for batch_caller_test

* Bump feeds

* Bump Solana

KuphJr committed:
FunctionsCoordinator v1.3.1 (#13815)

* FunctionsCoordinator v1.3.1

* fix: add NPM token to fetch private repo

* fix: use npm config set

* revert

* add auth credentials

* add auth credentials

* remove auth for npm

---------

Co-authored-by: Pablo <pablo.estrada@smartcontract.com>

jinhoonbang committed:
add empty bal spec (#13839)

* add empty bal spec

* add bal_spec_id

* include ccip spec ids

* remove orm changes

* add changeset

* add tag

---------

Co-authored-by: Chris <104409744+vreff@users.noreply.github.com>

erikburt committed:
fix: delete unused artfiacts during flakey test execution (#13842)

lukaszcl committed:
Bump CTF (#13846)

cedric-cordenier committed:
Merge remote-tracking branch 'origin/logging-follow-up' into logging-follow-up
